src/main/java/com/facebook/presto/pinot/grpc/GrpcRequestBuilder.java

```diff
@@ -0,0 +1,97 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.pinot.grpc;
+
+import com.facebook.presto.pinot.PinotScatterGatherQueryClient;
+import org.apache.pinot.common.proto.Server;
+import org.apache.pinot.common.request.BrokerRequest;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class GrpcRequestBuilder
+{
+    private int requestId;
+    private String brokerId = "unknown";
+    private boolean enableTrace;
+    private boolean enableStreaming;
+    private String payloadType;
+    private String sql;
+    private BrokerRequest brokerRequest;
+    private List<String> segments;
+
+    public GrpcRequestBuilder setRequestId(int requestId)
+    {
+        this.requestId = requestId;
+        return this;
+    }
+
+    public GrpcRequestBuilder setBrokerId(String brokerId)
+    {
+        this.brokerId = brokerId;
+        return this;
+    }
+
+    public GrpcRequestBuilder setEnableTrace(boolean enableTrace)
+    {
+        this.enableTrace = enableTrace;
+        return this;
+    }
+
+    public GrpcRequestBuilder setEnableStreaming(boolean enableStreaming)
+    {
+        this.enableStreaming = enableStreaming;
+        return this;
+    }
+
+    public GrpcRequestBuilder setSql(String sql)
+    {
+        payloadType = Constants.Request.PayloadType.SQL;
+        this.sql = sql;
+        return this;
+    }
+
+    public GrpcRequestBuilder setBrokerRequest(BrokerRequest brokerRequest)
+    {
+        payloadType = Constants.Request.PayloadType.BROKER_REQUEST;
+        this.brokerRequest = brokerRequest;
+        return this;
+    }
+
+    public GrpcRequestBuilder setSegments(List<String> segments)
+    {
+        this.segments = segments;
+        return this;
+    }
+
+    public Server.ServerRequest build()
+    {
+        if (payloadType == null || segments.isEmpty()) {
+            throw new PinotScatterGatherQueryClient.PinotException(PinotScatterGatherQueryClient.ErrorCode.PINOT_UNCLASSIFIED_ERROR, "Query and segmentsToQuery must be set");
+        }
+        Map<String, String> metadata = new HashMap<>();
+        metadata.put(Constants.Request.MetadataKeys.REQUEST_ID, Integer.toString(requestId));
+        metadata.put(Constants.Request.MetadataKeys.BROKER_ID, brokerId);
+        metadata.put(Constants.Request.MetadataKeys.ENABLE_TRACE, Boolean.toString(enableTrace));
+        metadata.put(Constants.Request.MetadataKeys.ENABLE_STREAMING, Boolean.toString(enableStreaming));
+        metadata.put(Constants.Request.MetadataKeys.PAYLOAD_TYPE, payloadType);
+
+        if (payloadType.equals(Constants.Request.PayloadType.SQL)) {
+            return Server.ServerRequest.newBuilder().putAllMetadata(metadata).setSql(sql).addAllSegments(segments)
+                    .build();
+        }
+        throw new RuntimeException("Unsupported payload type exception while serializing broker request: " + brokerRequest);
```

Comment:
This BrokerRequest is used by PinotBroker since we already parsed the query to BrokerRequest.
Here since we reconstruct the SQL, so I don't want to expose the BROKER_REQUEST type for now.